### DISCUSSION TITLE
Is there any way to save the parameter of the autotuning ?

### DISCUSSION THREAD

**Discussion by chenjunweii** [apache/incubator-mxnet#14539] [Q&A]
Is there any way to save the parameter of the autotuning so it won't have to find the best algorithms every time when program is launched ? thanks !

**Comment by haojin2** [comment]
Hi, which autotuning are you talking about? Is it something like "Finding the best convolution algorithm..."?

**Comment by chenjunweii** [comment]
@haojin2 yes, that is what i mean, the program autotune every time when it is launched, even though the program is not modified

**Comment by haojin2** [comment]
Well it's kind of a necessary step for now if you're using CuDNN, TF also has this tuning part for the CuDNN version. It's finding the fastest convolution algorithm given the memory constraints of your GPU. I think what you just said (caching the tuning result on the same machine) may be a way to save some time, I'll take a look into this ASAP and get back to you.

**Comment by chenjunweii** [comment]
@haojin2 Thanks ! and I know that tvm got something similar function which can tune conv layer or fullyconnected layer and save the tuned result to the model, so it can be loaded next time without tuning again. but such function seems not exist in mxnet.

**Comment by marcoabreu** [comment]
@ptrendx @dickjc123 @kellensunderland do you have an idea here?

**Comment by ThomasDelteil** [comment]
I think @KellenSunderland had a solution for that, see my issue here: https://github.com/apache/incubator-mxnet/issues/10567 that would definitely be helpful

**Comment by KellenSunderland** [comment]
Yes we've got a solution we use internally but it's not fleshed out enough to open source.  Could we possibly run a graph pass then save the graph with the tuning result serialized in an aux param or something?  (Then pick up the param results if we use that version of the graph).
